backend/tests/v5/scripts/modelProcessing/checkProcessingFlags.test.js

```diff
@@ -0,0 +1,136 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const {
+	determineTestGroup,
+	db: { reset: resetDB, createModel },
+	generateRandomString,
+	generateRandomModel,
+	generateRandomNumber,
+} = require('../../helper/services');
+const { times } = require('lodash');
+const { utilScripts, src } = require('../../helper/path');
+
+const { findModels } = require(`${src}/models/modelSettings`);
+const { deleteIfUndefined } = require(`${src}/utils/helper/objects`);
+const { disconnect } = require(`${src}/handler/db`);
+
+jest.mock('../../../../src/v5/services/mailer');
+const Mailer = require(`${src}/services/mailer`);
+
+const CheckProcessingFlags = require(`${utilScripts}/modelProcessing/checkProcessingFlags`);
+
+const modelStates = ['processing', 'queued'];
+
+const recentDate = new Date();
+// const oldDate = new Date();
+
+const setupData = () => {
+	const modelProms = times(2, async () => {
+		const teamspace = generateRandomString();
+		const models = await Promise.all(times(modelStates.length, async (n) => {
+			const { _id, name, properties } = generateRandomModel({
+				properties: deleteIfUndefined({ status: modelStates[n], timestamp: recentDate }),
```

Comment:
you should really have tests for every possible state, including undefined, and variation of date/time